File: src/main/java/com/dh/ondot/schedule/api/request/PlaceDto.java
package com.dh.ondot.schedule.api.request;

import com.dh.ondot.schedule.domain.Place;
import jakarta.validation.constraints.DecimalMax;
import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

public record PlaceDto(
        String title,

        @NotBlank String roadAddress,

        @NotNull @DecimalMin("-180.0") @DecimalMax("180.0") Double longitude,

        @NotNull @DecimalMin("-90.0") @DecimalMax("90.0") Double latitude
) {
    public static PlaceDto from(Place place) {
        return new PlaceDto(place.getTitle(), place.getRoadAddress(), place.getLongitude(), place.getLatitude());
    }
}

File: src/main/kotlin/com/dh/ondot/schedule/presentation/request/PlaceDto.kt
package com.dh.ondot.schedule.presentation.request

import com.dh.ondot.schedule.domain.Place
import jakarta.validation.constraints.DecimalMax
import jakarta.validation.constraints.DecimalMin
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull

data class PlaceDto(
    @field:NotBlank
    val title: String,

    @field:NotBlank
    val roadAddress: String,

    @field:NotNull @field:DecimalMin("-180.0") @field:DecimalMax("180.0")
    val longitude: Double,

    @field:NotNull @field:DecimalMin("-90.0") @field:DecimalMax("90.0")
    val latitude: Double,
) {
    companion object {
        fun from(place: Place): PlaceDto {
            return PlaceDto(place.title, place.roadAddress, place.longitude, place.latitude)
        }
    }
}
